contracts/LoanFactory.sol

```diff
@@ -0,0 +1,386 @@
+        onlyWorker
+    {
+        require(status == loan_status.ACTIVE);
+        require(now >= created_time + tenor);
+        require(transfer_records.length == transfer_records_id);
+        status = loan_status.MATURED;
+        emit Transfer(borrower_user_id, escrow_user_id, principal_amount, principal_currency, transfer_records_id++, "mature");
+    }
+
+    function mature_default()
+        external
+        onlyWorker
+    {
+        require(status == loan_status.MATURED);
+        require(now >= created_time + tenor + mature_lead_time);
+        require(transfer_records[transfer_records_id - 1] == default_transfer_record);
+        status = loan_status.PRINCIPAL_REPAYMENT_DEFAULT;
+        emit Transfer(escrow_user_id, liquidator_user_id, collateral_amount, collateral_currency, transfer_records_id++, "mature_default");
+    }
+
+    function complete()
+        external
+        onlyWorker
+    {
+        require(transfer_records.length == transfer_records_id);
+        require(transfer_records[transfer_records_id - 1] != default_transfer_record);
+        if (status == loan_status.MATURED) {
+            status = loan_status.COMPLETED;
+        } else if (status == loan_status.PRINCIPAL_REPAYMENT_DEFAULT) {
+            status = loan_status.COMPLETED;
+        } else {
+            revert();
+        }
+    }
+
+    constructor (
+        uint256[10] _uint_input,
+        bytes32[8] _bytes32_input,
+        address _owner,
+        address _worker
+    )
+        public
+    {
+        tenor = _uint_input[0];
+        principal_amount = _uint_input[1];
+        collateral_amount = _uint_input[2];
+        created_time = _uint_input[3];
+        lower_required_margin = _uint_input[4];
+        higher_required_margin = _uint_input[5];
+        margin_lead_time = _uint_input[6];
+        last_margin_time = _uint_input[7];
+        mature_lead_time = _uint_input[8];
+        interest_lead_time = _uint_input[9];
+        borrower_user_id = _bytes32_input[0];
+        holding_user_id = _bytes32_input[1];
+        escrow_user_id = _bytes32_input[2];
+        liquidator_user_id = _bytes32_input[3];
+        id = _bytes32_input[4];
+        order_id = _bytes32_input[5];
+        principal_currency = _bytes32_input[6];
+        collateral_currency = _bytes32_input[7];
+        owner = _owner;
+        worker = _worker;
+    }
+
+    function () external {
+        revert();
+    }
+}
+
+contract LoanFactory is Ownable {
+
+    address public worker;
+    address[] public loans;
+    uint256 public loan_id;
+
+    modifier onlyWorker() {
+        require(msg.sender == worker);
+        _;
+    }
+
+    event New_loan(address indexed loan, uint256 loan_id);
+
+    constructor () public {
+        worker = msg.sender;
+        loans.length = 1;
+    }
+
+    function change_worker(address _worker)
+        external
+        onlyOwner
+    {
+        worker = _worker;
+    }
+
+    function new_loan(
+        uint256[10] _uint_input,
```

Comment:
This inputs name are too generic, it is difficult to follow.
There is no reason to use underscore in front of the name here, right?
In addition, we need to keep consistency with the naming convention.
From reading https://solidity.readthedocs.io/en/latest/style-guide.html it looks like variable are lowercase separated by underscore, like you have done, but function names are **camelCased**. Therefore I would rename this as **newLoan**